Support target versions as strings

existentialism opened the pull request:
First part of #187.

Wanted to keep it relatively small, so only the following changes:

- Added `semverify` and moved `_extend` to utils file ala https://github.com/babel/babel-preset-env/pull/114
- `build-data` now outputs string values for env versions
- String target versions are now allowed via using `semver` to check if a plugin is required
- Left backwards compatibility (though cases like plugin `"node": 6.9` and target `"node": 6.10` will behave differently) with numeric target versions, not sure if we should drop in 2.0?

We should follow this up with https://github.com/babel/babel-preset-env/pull/114 then possibly add range support to targets (`"chrome": "^52"`) if we want.

yavorsky commented:
Maybe semverify all targets' versions immediately? If there are no difference between '5' and 5, why we should log it as a different values? Also it would be good for the `flow` brunch to have the same version type during the whole preset process. 😏

existentialism commented:
"Few" more changes:

- Normalizing (into semver) version now occur at `getTargets`, so they're strings all the way down.
- Moved `getTargets` into `targets-parser`, and refactored it a bit.
- `isPluginRequired` now assumes passed environments have already been parsed by `getTargets`, so it doesn't do checks or run `getTargets` itself
- For cleaner debugging output, added some utils to drop extraneous `.0` from versions

existentialism commented on scripts/build-data.js:
```diff
@@ -245,20 +245,22 @@ const generateData = (environments, features) => {
       const version = getLowestImplementedVersion(options, env);
       if (version !== null) {
-        plugin[env] = version;
+        plugin[env] = version.toString();
```
Should we `semverify` these?
